migrate agent options docs content to agent configuration page

mike-j-thomas opened the pull request:
See https://github.com/fleetdm/fleet/issues/13561 for context.

Changes:

- Created new docs page `agent-configuration.md` and migrated `agent options` content from https://fleetdm.com/docs/configuration/configuration-files#agent-options
- Added stub linking to the new page in the original agent options location.
- Moved content "Update agent options" out of [fleet-ui.md](https://fleetdm.com/docs/using-fleet/fleet-ui)
- ~Renamed `fleet-ui.md` to `queries.md` to reflect its content and set up a redirect `'GET /docs/using-fleet/fleet-ui': '/docs/using-fleet/queries',`~
- ~Added and updated `NavSection` and `pageOrderInSection` meta tags to reflect new navigation structure.~
- Updated [docs/Configuration/README.md](https://github.com/fleetdm/fleet/compare/docs-migrate-agent-options-to-agent-configuration-page?expand=1#diff-acd0a2b368992abbb3fd7b55a7a159ad824706e27e24aebed50a2d48e87b5ae2) to include a link and description for the new agent-configuration page.

mike-j-thomas commented:
@eashaw, there's a link in the "Agent options" page of "Organization settings" in the Fleet product that points to the docs content that is being relocated as part of this PR.

I spotted this in the Markdown, but not sure whether it is enough to stop the link from breaking (I suspect that it's not). Please confirm, thanks.

<img width="689" alt="image" src="https://github.com/fleetdm/fleet/assets/78363703/c09fa495-ebfa-49b3-8aa8-13a489086977">

cc @zhumo

rachaelshaw commented on docs/Configuration/configuration-files/README.md:
```diff
@@ -1488,3 +1118,4 @@ If you're using Fleet Premium, this enforces disk encryption on all hosts assign
 > **Note:** More settings are included in the [contributor documentation](https://github.com/fleetdm/fleet/blob/main/docs/Contributing/Configuration-for-contributors.md). It's possible, although not recommended, to configure these settings in the YAML configuration file.
 
 <meta name="description" value="Learn how to use configuration files and the fleetctl command line tool to configure Fleet.">
+<meta name="navSection" value="Fleet Server">
```
```suggestion
<meta name="navSection" value="Fleet server">
```

rachaelshaw commented on docs/Configuration/fleet-server-configuration.md:
```diff
@@ -3228,3 +3228,4 @@ If the IP retrieved using the above heuristic belongs to a private range, then F
 
 <meta name="pageOrderInSection" value="100">
 <meta name="description" value="This page includes resources for configuring the Fleet binary, managing osquery configurations, and running with systemd.">
+<meta name="navSection" value="Fleet Server">
```
```suggestion
<meta name="navSection" value="Fleet server">
```

mike-j-thomas commented:
> @eashaw, there's a link in the "Agent options" page of "Organization settings" in the Fleet product that points to the docs content that is being relocated as part of this PR.
> 
> I spotted this in the Markdown, but not sure whether it is enough to stop the link from breaking. The content is being moved from `fleet-ui.md` to `agent-configuration.md`.
> 
> Please confirm, thanks.
> 
> <img alt="image" width="689" src="https://user-images.githubusercontent.com/78363703/263998096-c09fa495-ebfa-49b3-8aa8-13a489086977.png">
> EDIT: I realize now that the Markdown is because the hashlink changed in the docs, but redirects cannot pick up the hash link, so it will break 😞 (https://fleetdm.com/docs/using-fleet/fleet-ui#configuring-agent-options)
> 
> cc @zhumo

@eashaw, no action is needed here. I have reverted my proposed changes for the Fleet UI docs page. The content strategy for that page requires designing more thoroughly and is out of the scope of this PR.

@rachaelshaw, I've moved this PR out of draft, and it is ready for your review.
